src/Analysis/Engine/Impl/Values/BuiltinClassInfo.cs

```diff
@@ -158,7 +159,22 @@ public BuiltinClassInfo(IPythonType classObj, PythonAnalyzer projectState)
         }
 
         public override IAnalysisSet GetIndex(Node node, AnalysisUnit unit, IAnalysisSet index) {
-            // TODO: Needs to actually do indexing on type
+            try {
+                var indices = index.Select(x => x.GetConstantValue()).ExcludeDefault().OfType<int>().ToArray();
+                if (this is IPythonSequenceType seqt) {
+                    var a = seqt.IndexTypes.ToArray();
+                    var values= indices.Where(i => i >= 0 && i < a.Length).Select(i => unit.State.GetAnalysisValueFromObjects(a[i]));
+                    return AnalysisSet.UnionAll(values);
+                }
+                if (this is IBuiltinSequenceClassInfo bseqt) {
+                    var a = bseqt.IndexTypes.ToArray();
```

Comment:
Shouldn't we move this implementation into `GetIndex` override in `SequenceBuiltinClassInfo`?

Reply:
Yes